hr_expense_receipt_require/models/hr_expense.py

```diff
@@ -10,23 +10,31 @@ class HrExpense(models.Model):
     @api.multi
     def action_submit_expenses(self):
         for rec in self:
-            if rec.product_id.expense_receipt_required and \
-                    not rec.attachment_number:
-                raise UserError(_('You need to provide a receipt'
-                                  ' to submit this expense!'))
+            if rec.product_id.expense_receipt_required and not rec.attachment_number:
```

Comment:
This could raise the validation error even if the `allow_without_attachment` is set.
It is best to consider the Expense Report flag, just in case.
```suggestion
            if rec.product_id.expense_receipt_required and not rec.attachment_number and not sheet_id.allow_without_attachment :
```

Reply:
Made this change as:
rec.sheet_id.allow_without_attachment

Reply:
Yes, that looks right, thank you.

Reply:
Thank you